enhancements/network/ovn-kubernetes-evpn.md

```diff
@@ -0,0 +1,292 @@
+enabled.
+
+The EVPN feature is only supported when `routingViaHost` is set to false, also
+known as local gateway mode. CNO will perform no validation, OVN-Kubernetes will
+be in charge of reporting invalid configurations in this regard.
+
+The EVPN feature requires FRR-k8s. FRR-k8s is deployed by CNO when the
+`additionalRoutingCapabilities` `providers` includes `FRR` in the operator
+configuration, which is required to enable `routeAdvertisements`.
+
+The upstream enhancement may introduce changes to the FRR-k8s APIs as well.
+These API changes need not to be gated by the feature gate introduced above.
+Worth noting that FRR-k8s APIs allow the possibility to inject raw FRR
+configuration and that this capability might be used during the development
+process until proper structured APIs are introduced.
+
+Currently, CNO FRR-k8s includes FRR v8 but the EVPN feature needs FRR v9+ so it
+will need to be updated. The problem is that FRR-k8s consumes the FRR version
+provided by RHEL, specifically RHEL9 which is the current version in use. RHEL9
+only provides FRR v8. RHEL10 will provide FRR v10. RHEL10 was to be
+[available][7] as tech preview in OCP 4.21 and fully available in OCP 4.22 but
+currently its real timeline is uncertain. These are our options in order of
+preference:
+
+[7]: https://docs.google.com/spreadsheets/d/1VO00pWkWf8Fr30PHl8mZFTK9ZnJO51BGXH4FT6efwp4/edit?gid=1551125754#gid=1551125754
+
+* Request RHEL to package FRR v10 for RHEL 9.
+* If RHEL 10 is available, build a FRR-k8s image against a RHEL10 stream.
+* Do our own build of FRR v10 for RHEL 9 and cross tag it for OCP as we do with
+  OVN. This will require to assess and cover testing gaps vs the official RHEL
+  build.
+
+A user of the EVPN feature may leverage the internal BGP fabric introduced in
+the No-Overlay mode [enhancement][8], particularly for East/West L2 EVPN
+configurations. No additional changes with respect to what is described in that
+enhancement are required.
+
+[8]: https://github.com/openshift/enhancements/pull/1859
+
+
+### Workflow Description
+
+In no particular order, a cluster administrator enables FRR and
+RouteAdvertisements, and the EVPN feature gate if not available through the
+default feature set:
+
+```shell
+oc patch featuregate cluster --type=merge -p='{"spec"{"featureSet":"TechPreviewNoUpgrade"}'
+...
+oc patch Network.operator.openshift.io cluster --type=merge -p='{"spec":{"additionalRoutingCapabilities": {"providers": ["FRR"]}, "defaultNetwork":{"ovnKubernetesConfig":{"routeAdvertisements":"Enabled"}}}}'
+```
+
+Then, a cluster administrator enables the internal BGP fabric if intended to be
+used:
+```shell
+TODO
+```
+
+Then, a cluster administrator follows the OVN-Kubernetes [workflow][9] to configure EVPN.
+
+[9]: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/8461a5526377488bc643bd4eb7024b0b735e8830/docs/okeps/okep-50088-evpn.md#workflow-description
+
+
+### API Extensions
+
+There are no changes required to OCP specific APIs.
```

Comment:
but we want the API team to provide feedback on the ovn-k / frr-k8s APIs that are added / updated, right ? 

Where are they supposed to provide feedback ? In the original ovn-k enhancement ? 

I think we should try to capture all the required API changes in this document, to get an holistic view on how to configure the feature in a single place.